gnucash/gnome-utils/gnc-gui-query.c

```diff
@@ -315,3 +315,98 @@ gnc_choose_radio_option_dialog(GtkWidget *parent,
+gchar *
+gnc_input_dialog (GtkWidget *parent, const gchar *title, const gchar *msg, const gchar *default_input)
+{
+    gint result;
+    GtkWidget *view;
+    GtkTextBuffer *buffer;
+    gchar *user_input = NULL;
+    GtkTextIter start, end;
+    
+    /* Create the widgets */
+    GtkWidget* dialog = gtk_dialog_new_with_buttons (title, GTK_WINDOW (parent),
+                                          GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
+                                          _("_OK"), GTK_RESPONSE_ACCEPT,
+                                          _("_Cancel"), GTK_RESPONSE_REJECT,
+                                          NULL);
+    GtkWidget* content_area = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+    
+    // add a label
+    GtkWidget* label = gtk_label_new (msg);
+    gtk_box_pack_start(GTK_BOX(content_area), label, FALSE, FALSE, 0);
+    
+    // add a textview
+    view = gtk_text_view_new ();
+    gtk_text_view_set_wrap_mode (GTK_TEXT_VIEW (view), GTK_WRAP_WORD_CHAR);
+    buffer = gtk_text_view_get_buffer (GTK_TEXT_VIEW (view));
+    gtk_text_buffer_set_text (buffer, default_input, -1);
+    gtk_box_pack_start(GTK_BOX(content_area), view, TRUE, TRUE, 0);
+    
+    // run the dialog
+    gtk_widget_show_all (dialog);
+    result = gtk_dialog_run (GTK_DIALOG (dialog));
+    
+    if (result != GTK_RESPONSE_REJECT)
+    {
+        gtk_text_buffer_get_start_iter (buffer, &start);
+        gtk_text_buffer_get_end_iter (buffer, &end);
+        user_input = gtk_text_buffer_get_text (buffer,
+                                               &start, &end, FALSE);
+    }
+    
+    gtk_widget_destroy (dialog);
+    
+    return user_input;
+}
+
+void
+gnc_info2_dialog (GtkWidget *parent, const gchar *title, const gchar *msg)
+{
+    GtkWidget *view;
+    GtkTextBuffer *buffer;
+    gint width, height;
+    
+    /* Create the widgets */
+    GtkWidget* dialog = gtk_dialog_new_with_buttons (title, GTK_WINDOW (parent),
+                                          GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
+                                          _("_OK"), GTK_RESPONSE_ACCEPT,
+                                          NULL);
+    GtkWidget* content_area = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+    
+    // add a scroll area
+    GtkWidget* scrolledwindow = gtk_scrolled_window_new (NULL, NULL);
+    gtk_box_pack_start(GTK_BOX(content_area), scrolledwindow, TRUE, TRUE, 0);
+    
+    // add a textview
+    view = gtk_text_view_new ();
+    gtk_text_view_set_editable (GTK_TEXT_VIEW (view), FALSE);
+    buffer = gtk_text_view_get_buffer (GTK_TEXT_VIEW (view));
+    gtk_text_buffer_set_text (buffer, msg, -1);
+    gtk_container_add (GTK_CONTAINER (scrolledwindow), view);
+    
+    // run the dialog
+    if (parent)
+    {
+        gtk_window_get_size (GTK_WINDOW(parent), &width, &height);
+        gtk_window_set_default_size (GTK_WINDOW(dialog), width, height);
+    }
```

Comment:
I think it's generally a good idea to ensure `parent` is never `NULL`. So, earlier on we should have `g_return_if_fail (parent);` and audit callers of this code. This is a good time to fix this.

Reply:
`static_assert`, not `g_return_if_fail`. We want the failure at compile time, not tripping up users at run time.

Reply:
But that should be a separate PR that does the same in all dialogs.